src/js/containers/agencyV2/accountSpending/CountTabContainer.jsx

```diff
@@ -3,60 +3,70 @@
  * Created by Lizzie Salita 5/11/20
  */
 
-import React, { useState, useEffect } from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
+import { camelCase } from 'lodash';
+import { useSelector, useDispatch } from 'react-redux';
+import { Tabs } from 'data-transparency-ui';
 import { fetchSpendingCount } from 'apis/agencyV2';
-import CountTab from 'components/agencyV2/CountTab';
+import {
+    setBudgetCategoryCount,
+    resetBudgetCategoryCounts
+} from 'redux/actions/agencyV2/agencyV2Actions';
 
 const propTypes = {
     fy: PropTypes.string.isRequired,
     agencyId: PropTypes.string.isRequired,
-    type: PropTypes.string.isRequired,
-    label: PropTypes.string.isRequired,
-    subHeading: PropTypes.string,
+    tabs: PropTypes.arrayOf(PropTypes.shape({
+        internal: PropTypes.string.isRequired,
+        label: PropTypes.string.isRequired,
+        countField: PropTypes.string.isRequired,
+        subHeading: PropTypes.string
+    })),
     setActiveTab: PropTypes.func.isRequired,
-    active: PropTypes.bool,
-    countField: PropTypes.string,
-    subCountField: PropTypes.string
+    activeTab: PropTypes.string.isRequired
 };
 
 const CountTabContainer = (props) => {
-    const [count, setCount] = useState(null);
-    const [subCount, setSubCount] = useState(null);
-    const [isLoading, setIsLoading] = useState(false);
```

Comment:
Why have these in Redux instead of local state? Or should we use the table details to store the count in Redux, then use that here instead of another API request?

Reply:
Good questions. I put them in Redux because I think it's a better solution for nested state than hooks. Component state would require a custom hook (that would more or less be a reducer) or three different state variables (`programActivityCount`, `objectClassCount` etc.) and keeping track of loading/ error state for each. 
Page metadata would also give us the counts, but we need counts for every tab on initial load of the page and only need the table data once a user clicks on a tab other than the first one.

Reply:
Valid points; approved